appinventor/appengine/src/com/google/appinventor/client/Ode.java

```diff
@@ -1526,15 +1523,17 @@ public void onClick(Widget sender) {
    * are present.
    */
   private void maybeShowNoProjectsDialog() {
-    projectManager.addProjectManagerEventListener(new ProjectManagerEventAdapter() {
-      @Override
-      public void onProjectsLoaded() {
-        if (!ProjectListBox.getProjectListBox().getProjectList().listContainsProjects() && !templateLoadingFlag &&
-          !newGalleryLoadingFlag) {
-          ErrorReporter.hide();  // hide the "Please choose a project" message
-          createNoProjectsDialog(true);
+    projectManager.loadProjects(projectService).then0(() -> {
+      for (Project p : projectManager.getProjects()) {
```

Comment:
@ewpatton This resolves the problem I reported. However, what is happening here is not obvious and a comment might be a good addition! In particular, the fact that after the first call to loadProjects it may return an incorrect project list because it is using the original list from the initial call at system load. However, this is compensated for by using the in-memory list of projects (projectManager.getProjects()) that is up-to-date.

Reply:
I agree. I am planning to rename the function to something along the lines of `ensureProjectsLoadedFromServer` so that it's clear that this is a precondition. My change to using `getProjects()` is more in line with the logic used in production today.